src/main/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -1265,4 +1267,26 @@ public static boolean isValidUnsignedIntegerAttribute(long value) {
     public static boolean isReferenceSequenceIncompatibleWithBAI(final SAMSequenceRecord sequence) {
         return sequence.getSequenceLength() > GenomicIndexUtil.BIN_GENOMIC_SPAN;
     }
+
+    public static String calculateOATagValue(SAMRecord record){
+        if (record.getReferenceName().contains(",")) {
+            throw new SAMException(String.format("Reference name for record %s contains a comma character.", record.getReadName()));
+        }
+        final String oaValue;
+        if (record.getReadUnmappedFlag()) {
+            oaValue = String.format("*,0,%s,*,%s,",
+                    record.getReadNegativeStrandFlag() ? Strand.NEGATIVE : Strand.POSITIVE,
+                    record.getMappingQuality());
+        } else {
+            oaValue = String.format("%s,%s,%s,%s,%s,%s",
+                    record.getReferenceName(),
+                    record.getAlignmentStart(),
+                    record.getReadNegativeStrandFlag() ? Strand.NEGATIVE : Strand.POSITIVE,
+                    record.getCigarString(),
+                    record.getMappingQuality(),
+                    Optional.ofNullable(record.getAttribute(SAMTag.NM.name())).orElse(""));
```

Comment:
Annoyingly this actually creates a new `Optional` object which adds needless overhead.  A ternery expression is actually more performant here I believe.  That said, this overhead is probably negligible and unlikely to ever show up in a profiler ever.  The string formatter probably has a lot more overhead than an extra Optional.